packages/builtin_test/_test.pony

```diff
@@ -1764,6 +1765,29 @@ class \nodoc\ iso _TestArrayFromCPointer is UnitTest
     let arr = Array[U8].from_cpointer(Pointer[U8], 1, 1)
     h.assert_eq[USize](0, arr.size())
 
+class \nodoc\ iso _TestArrayCopyTo is UnitTest
+  fun name(): String => "builtin/Array.copy_to_with_uninitialized_source"
+
+  fun apply(h: TestHelper) =>
+    // Test that a using an uninitialized array as a source leaves the destination unchanged
+    let src1: Array[U8] = []
+    let dest1: Array[U8] = [0; 1; 2; 3; 4; 5; 6]
+    src1.copy_to(dest1, 0, 0, 10)
+    h.assert_array_eq[U8]([0; 1; 2; 3; 4; 5; 6], dest1)
+
+    // Test that copying from an empty source array leaves 
+    // the destination unchanged
+    let dest2: Array[U8] = [0; 1; 2; 3; 4; 5; 6]
+    try
+      // create an initialized array and then remove it's only element
+      // leaving it empty
+      let src2: Array[U8] = [1]
+      src2.pop()?
```

Comment:
so this is an amusing little test I think it is best to put the try only around src2.pop()?

like `try src2.pop()? end`.

If the test initialization gets messed up, doing the try like that will catch the problem, otherwise we will be silently skipping any actually meaningful test.

it would also be reasonable to assert after the pop (outside the try) that the size of src2 array is 0. 

Can you make those changes?